TrackingPerformance/Condor/condorJobs_sim.py

```diff
@@ -1,156 +1,248 @@
+    )
+    return parser.parse_args()
+
+
+def load_config(config_name):
+
+    config_file = Path(config_name)
+
+    # Ensure the config file has the .py extension
+    if config_file.suffix != ".py":
+        config_file = config_file.with_suffix(".py")
+
+    # Create an absolute path
+    config_path = config_file.resolve()
+
+    # Ensure the file exists
+    if not config_path.exists():
+        raise FileNotFoundError(f"The configuration file {config_path} does not exist.")
+
+    spec = importlib.util.spec_from_file_location("config", config_path)
+    config = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(config)
+    return config
+
+
+def main():
+
+    args = parse_args()
+    config = load_config(args.config)
+
+    # ==========================
+    # Check paths
+    # ==========================
+
+    assert (
+        config.sim_steering_file.exists()
+    ), f"The file {config.sim_steering_file} does not exist"
+    assert (
+        config.rec_steering_file.exists()
+    ), f"The file {config.rec_steering_file} does not exist"
+    assert (
+        config.detectorDIR.exists()
+    ), f"The folder {config.detectorDIR} does not exist"
+
+    # ==========================
+    # Parameters Initialisation
+    # ==========================
+
+    # Define lists of parameters for reconstruction
+    thetaList_ = config.thetaList_
+    momentumList_ = config.momentumList_
+    particleList_ = config.particleList_
+
+    DetectorModelList_ = config.detectorModel
+    Nevts_ = config.Nevts_
+    Nevt_per_job = config.Nevt_per_job  # Set the desired number of events per job
+
+    total_events = int(Nevts_)
+    N_para_sets = (
+        len(DetectorModelList_)
+        * len(particleList_)
+        * len(thetaList_)
+        * len(momentumList_)
+    )
+    N_jobs_per_para_set = total_events // int(
+        Nevt_per_job
+    )  # number of parallel jobs with same parameter combination/set
+    N_jobs = N_jobs_per_para_set * N_para_sets  # total number of jobs
+
+    # ===========================
+    # Directory Setup and Checks
+    # ===========================
+
+    # Define environment setup path
+    environ_path = config.setup
+
+    # Define directories for input and output
+    directory_jobs = config.SIMcondorDir / f"{particleList_[0]}_{DetectorModelList_[0]}"
+    SIMEOSDir = config.dataDir / f"{DetectorModelList_[0]}" / "SIM"  # output
+
+    # Enable output checks
+    CHECK_OUTPUT = True  # Set to True to enable checks, False to disable
+    # It will check if the ouputs exist and contain correct number of events
+    # if not it will send job to rerun simulation
+
+    JOB_FLAVOR = "testmatch"
+    # Job flavours:
+    #   espresso     = 20 minutes
+    #   microcentury = 1 hour
+    #   longlunch    = 2 hours
+    #   workday      = 8 hours
+    #   tomorrow     = 1 day
+    #   testmatch    = 3 days
+    #   nextweek     = 1 week
```

Comment:
I know it was already `testmatch` before but does it really take that long? I think we might want to aim for job sizes closer to `longlunch`

We also want to be able to set this from the config.

Reply:
Fair point. Done